packages/components/src/tree-grid/index.js

```diff
@@ -91,8 +91,12 @@ function TreeGrid(
 					);
 				}
 
-				// Focus is either at the left or right edge of the grid.
-				if ( nextIndex === currentColumnIndex ) {
+				// Focus is either at the left or right edge of the grid. Allow Right Arrow to expand the row.
+				if (
+					nextIndex === currentColumnIndex ||
+					( activeRow.hasAttribute( 'aria-expanded' ) &&
```

Comment:
Instead of checking the index, I opted to check the activeRow to see if it contained aria-expanded or not. This in theory should be more compatible with possible future uses as it doesn't require a hard coded index value.